packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -208,9 +209,10 @@ class SyncEngineImpl implements SyncEngine {
   queriesByTarget = new Map<TargetId, Query[]>();
   /**
    * The keys of documents that are in limbo for which we haven't yet started a
-   * limbo resolution query.
+   * limbo resolution query. The strings in this set are the result of calling
+   * `key.path.canonicalString()` where `key` is a `DocumentKey` object.
    */
-  enqueuedLimboResolutions: DocumentKey[] = [];
+  enqueuedLimboResolutions = new Set<string>();
```

Comment:
We should maybe mention that we can use this datastructure in JS because it preserves insertion order.